packages/react-router-redux/examples/AuthExample.js

```diff
@@ -1,111 +1,115 @@
+import { createStore, applyMiddleware, combineReducers } from 'redux'
+import createHistory from 'history/createBrowserHistory'
 
-import { connect } from 'react-redux';
-import { Route, Switch } from 'react-router';
+import { Route, Switch } from 'react-router'
 
-const history = createHistory();
+const history = createHistory()
 
 const authSuccess = () => ({
-    type: 'AUTH_SUCCESS'
+  type: 'AUTH_SUCCESS'
 })
 
 const authFail = () => ({
-    type: 'AUTH_FAIL'
+  type: 'AUTH_FAIL'
 })
 
 const initialState = {
-    isAuthenticated: false
+  isAuthenticated: false
 }
 
 const authReducer = (state = initialState , action) => {
-    switch (action.type) {
-        case 'AUTH_SUCCESS':
-            return Object.assign({}, ...state, {
-                isAuthenticated: true
-            });
-        case 'AUTH_FAIL':
-            return Object.assign({}, ...state, {
-                isAuthenticated: false
-            });
-        default: 
-            return { ...state };
-    }
+  switch (action.type) {
+    case 'AUTH_SUCCESS':
+      return {
+        ...state,
+        isAuthenticated: true
+      }
+    case 'AUTH_FAIL':
+      return {
+        ...state,
+        isAuthenticated: false
+      }
+    default:
+      return state
+  }
 }
 
 const store = createStore(
-    combineReducers({ routerReducer, authReducer }),
-    applyMiddleware(routerMiddleware(history)),
-);
+  combineReducers({ routerReducer, authReducer }),
+  applyMiddleware(routerMiddleware(history)),
+)
 
 const ConnectedSwitch = connect(state => ({
 	location: state.location
-}))(Switch);
+}))(Switch)
 
 class LoginContainer extends React.Component {
-    render() {
-        return <button onClick={() => this.props.login()}>Login Here!</button>
-    }
+  render() {
+    return <button onClick={this.props.login}>Login Here!</button>
+  }
 }
 
 class HomeContainer extends React.Component {
-    componentWillMount() {
-        alert('Private home is at: ' + this.props.location.pathname)
-    }
+  componentWillMount() {
+    alert('Private home is at: ' + this.props.location.pathname)
+  }
 
-    render() {
-        return <button onClick={() => this.props.logout()}>Logout Here!</button>
-    }
+  render() {
+    return <button onClick={this.props.logout}>Logout Here!</button>
+  }
 }
 
 const AppContainer = ({ location }) => (
-    <ConnectedSwitch>
-        <PrivateRoute component={Home} />
-    </ConnectedSwitch>
-);
+  <ConnectedSwitch>
+    <PrivateRoute component={Home} />
+  </ConnectedSwitch>
+)
 
 class PrivateRouteContainer extends React.Component {
-    render() {
-        return this.props.isAuthenticated ?
-            <Route exact path="/" component={this.props.component} />
-        :   <Route exact path="/" component={Login} />
-    }
+  render() {
+    return this.props.isAuthenticated ?
+        <Route exact path="/" component={this.props.component} />
+    :   <Route exact path="/" component={Login} />
+  }
```

Comment:
Really liked what you did here, 😄  what do you think about using **react-router**'s _Redirect_ for auth flows instead of rendering two different _Routes_?

You can check an example [here](https://reacttraining.com/react-router/web/example/auth-workflow).

Reply:
Sounds great to me!